lib/phoenix/code_reloader/group_leader_proxy.ex

```diff
@@ -0,0 +1,64 @@
+# Spawned by the Proxy when capture is started,
+# to capture stdout of the compilation process.
+# Restores the original group leader when terminated after compilation.
+# Output is forwarded to the parent proxy with the :stdout message.
+defmodule Phoenix.CodeReloader.GroupLeaderProxy do
```

Comment:
Something like this:

```elixir
defp handle_io(msg, device) do
  case msg do
    {:io_request, ..., ...}
    _ ->
  end
end
```

then the group leader proxy is:

```
Task.start_link(fn ->
  replace group leader...
  loop("")
end)

defp loop(arg1, arg2, arg3) do
  receive do
    :stop -> ...
    msg ->
      buffer = handle_io(...)
      loop(buffer)
```

and the `handle_info` of the proxy also calls handle_io when necessary.